File: apps/mybookkeeper/backend/app/schemas/listings/listing_list_response.py
"""Paginated envelope for GET /listings.

Replaces the bare `list[ListingSummary]` response from PR 1.1b — without a
total + has_more flag the frontend's "Load more" button has no terminator
(known tech debt logged on PR 1.1b). This envelope mirrors the convention
used by other paginated routes and gives the client an authoritative end
signal.
"""
from __future__ import annotations

from platform_shared.schemas.pagination import ListResponse

from app.schemas.listings.listing_summary import ListingSummary


class ListingListResponse(ListResponse[ListingSummary]):
    """Paginated response envelope for GET /listings."""

File: apps/mybookkeeper/frontend/src/shared/types/listing/listing-list-response.ts
import type { ListingSummary } from "@/shared/types/listing/listing-summary";

/**
 * Paginated envelope returned by GET /listings.
 *
 * Replaces the bare `ListingSummary[]` shape from PR 1.1b. The `has_more`
 * flag lets the frontend hide the "Load more" button on the last page —
 * closes the pagination-terminator gap logged in TECH_DEBT.md.
 */
export interface ListingListResponse {
  items: ListingSummary[];
  total: number;
  has_more: boolean;
}
